corehq/apps/userreports/views.py

```diff
@@ -439,6 +441,28 @@ def preview_data_source(request, domain, config_id):
     return render(request, "userreports/preview_data.html", context)
 
 
+def process_url_params(params, column_descriptions):
+    columns = {column['name']: column for column in column_descriptions}
+    keyword_filters = {}
+    sql_filters = []
+    for key, value in params.items():
+        if key == 'format':
+            continue
+        column = columns.get(key, None)
+        if not column:
+            raise BadSpecError('Invalid filter parameter: {}'.format(key))
+
+        if (
+            value == 'last30'
+            and isinstance(column['type'], (types.Date, types.DateTime))
```

Comment:
I'd like to generalize this a bit to support multiple "meta" parameters, probably with some kind of mapping between url param and sql filter.  Might be a bit premature though.